lang/wasm_alien_thread_runner.cc

```diff
@@ -38,6 +41,12 @@ alien_thread_runner::alien_thread_runner()
         sigfillset(&mask);
         auto r = ::pthread_sigmask(SIG_BLOCK, &mask, nullptr);
         throw_pthread_error(r);
+
+        int nice_value = nice(10);
+        if (nice_value == -1 && errno != 0) {
```

Comment:
quote from nice(2):

>        On  success,  the new nice value is returned (but see NOTES below).  On
>        error, -1 is returned, and errno is set to indicate the error.
> 
>        A successful call can legitimately return -1.  To detect an error,  set
>        errno  to  0  before  the  call,  and check whether it is nonzero after
>        nice() returns -1.

so, i think, just to be on the safe side, if we really want to check for an error returned by `nice(2)`, we need to do something like https://github.com/coreutils/coreutils/blob/76f2fb627118a26c25003dbd98c22c153b7ee1d2/src/nice.c#L195-L198 , where the implementation of `nice(1)` command line tool resets `errno` before calling `nice(2)`, just to ensure that it always get an `errno` set by the call.

Reply:
I was actually trying to follow this manual, but forgot this step. Added in rebase